File: weather/src/main/java/com/nerdery/pkarels/weather/data/model/WeatherResponse.java
package com.nerdery.pkarels.weather.data.model;

import com.google.gson.annotations.SerializedName;

/**
 * Response from DarkSky weather requests in {@link com.nerdery.pkarels.weather.data.api.WeatherService}
 */
public class WeatherResponse {

    @SerializedName("currently")
    private ForecastCondition currentForecast;

    private HourlyResponse hourly;

    public WeatherResponse(ForecastCondition currentForecast, HourlyResponse hourly) {
        this.currentForecast = currentForecast;
        this.hourly = hourly;
    }

    /**
     * Current Weather Condition
     * @return ForecastCondition
     */
    public ForecastCondition getCurrentForecast() {
        return currentForecast;
    }

    /**
     * Hourly Response model that contains list of ForecastConditions
     * @return HourlyResponse
     */
    public HourlyResponse getHourly() {
        return hourly;
    }
}

File: weather/src/main/java/com/nerdery/pkarels/weather/model/WeatherResponse.kt
package com.nerdery.pkarels.weather.model

import com.google.gson.annotations.SerializedName

/**
 * Response from DarkSky weather requests in [com.nerdery.pkarels.weather.data.api.WeatherService]
 */
class WeatherResponse(@field:SerializedName("currently")
                      /**
                       * Current Weather Condition
                       * @return ForecastCondition
                       */
                      val currentForecast: ForecastCondition,
                      /**
                       * Hourly Response model that contains list of ForecastConditions
                       * @return HourlyResponse
                       */
                      val hourly: HourlyResponse) {
    lateinit var forecasts: DayForecasts
}
